In [1]:
from utils import load_finetuned_qwen
from title_search_evaluation import get_scores

/home/adamp/school/cmpt419/layman-scientific-embeddings/venv/lib/python3.12/site-packages/tqdm/auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html
  from .autonotebook import tqdm as notebook_tqdm


In [2]:
def print_model_scores(model):
    scores = get_scores(model)
    print(f"Full match score: {scores["perfect_match_score"] * 100:.2f}%")
    print(f"Related keywords score: {scores["in_top_results_score"] * 100:.2f}%")

Base scibert

In [3]:
print_model_scores("allenai/scibert_scivocab_uncased")

Loading weights: 100%|██████████| 199/199 [00:00<00:00, 22061.28it/s]
BertModel LOAD REPORT from: allenai/scibert_scivocab_uncased
Key                                        | Status     |  | 
-------------------------------------------+------------+--+-
cls.predictions.decoder.weight             | UNEXPECTED |  | 
cls.predictions.transform.LayerNorm.bias   | UNEXPECTED |  | 
cls.seq_relationship.weight                | UNEXPECTED |  | 
cls.predictions.transform.LayerNorm.weight | UNEXPECTED |  | 
cls.predictions.transform.dense.bias       | UNEXPECTED |  | 
cls.predictions.bias                       | UNEXPECTED |  | 
cls.predictions.decoder.bias               | UNEXPECTED |  | 
cls.seq_relationship.bias                  | UNEXPECTED |  | 
cls.predictions.transform.dense.weight     | UNEXPECTED |  | 

Notes:
- UNEXPECTED:	can be ignored when loading from different task/architecture; not ok if you expect identical arch.


Full match score: 1.98%
Related keywords score: 5.63%


Vanilla scibert

In [4]:
print_model_scores("models/vanilla-scibert")

Loading weights: 100%|██████████| 199/199 [00:00<00:00, 2946.65it/s]


Full match score: 5.83%
Related keywords score: 25.20%


Jargon-layman scibert

In [5]:
print_model_scores("models/jargon-layman-scibert-1.00")

Loading weights: 100%|██████████| 199/199 [00:00<00:00, 2815.59it/s]


Full match score: 6.13%
Related keywords score: 26.19%


Vanilla qwen

In [6]:
print_model_scores(
    load_finetuned_qwen("models/vanilla-qwen")
)

Loading weights:   0%|          | 1/310 [00:00<00:55,  5.57it/s]/home/adamp/school/cmpt419/layman-scientific-embeddings/venv/lib/python3.12/site-packages/bitsandbytes/backends/cuda/ops.py:213: FutureWarning: _check_is_size will be removed in a future PyTorch release along with guard_size_oblivious.     Use _check(i >= 0) instead.
  torch._check_is_size(blocksize)
Loading weights: 100%|██████████| 310/310 [00:01<00:00, 274.45it/s]


Full match score: 4.84%
Related keywords score: 17.09%


Jargon-layman qwen

In [7]:
print_model_scores(
    load_finetuned_qwen("models/jargon-layman-qwen")
)

Loading weights:   0%|          | 0/310 [00:00<?, ?it/s]/home/adamp/school/cmpt419/layman-scientific-embeddings/venv/lib/python3.12/site-packages/bitsandbytes/backends/cuda/ops.py:213: FutureWarning: _check_is_size will be removed in a future PyTorch release along with guard_size_oblivious.     Use _check(i >= 0) instead.
  torch._check_is_size(blocksize)
Loading weights: 100%|██████████| 310/310 [00:00<00:00, 1099.23it/s]


Full match score: 4.84%
Related keywords score: 17.09%


# Results

| Model | Exact keyword match score | Related keyword score |
|---|---|---|
| allenai/scibert_scivocab_uncased | 1.98% | 5.63% |
| vanilla-scibert | 5.83% | 25.20% |
| jargon-layman-scibert | 6.13% | 26.19% |
| vanilla-qwen | 4.84% | 17.09% |
| jargon-layman-qwen | 4.84% | 17.09% |